src/libraries/System.Private.CoreLib/src/System/SearchValues/IndexOfAnyAsciiSearcher.cs

```diff
@@ -602,6 +676,416 @@ private static unsafe bool TryLastIndexOfAny<TNegator>(ref short searchSpace, in
+        [CompExactlyDependsOn(typeof(Ssse3))]
+        [CompExactlyDependsOn(typeof(AdvSimd))]
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        public static int IndexOfAny<TNegator, TOptimizations>(ref short searchSpace, int searchSpaceLength, ref AsciiWithSecondSetState state)
+            where TNegator : struct, INegator
+            where TOptimizations : struct, IOptimizations =>
+            IndexOfAnyCore<int, TNegator, TOptimizations, IndexOfAnyResultMapper<short>>(ref searchSpace, searchSpaceLength, ref state);
+
+        [CompExactlyDependsOn(typeof(Ssse3))]
+        [CompExactlyDependsOn(typeof(AdvSimd))]
+        [CompExactlyDependsOn(typeof(PackedSimd))]
+        private static TResult IndexOfAnyCore<TResult, TNegator, TOptimizations, TResultMapper>(ref short searchSpace, int searchSpaceLength, ref AsciiWithSecondSetState state)
+            where TResult : struct
+            where TNegator : struct, INegator
+            where TOptimizations : struct, IOptimizations
+            where TResultMapper : struct, IResultMapper<short, TResult>
+        {
+            ref short currentSearchSpace = ref searchSpace;
+
+            if (searchSpaceLength < Vector128<ushort>.Count)
+            {
+                ref short searchSpaceEnd = ref Unsafe.Add(ref searchSpace, searchSpaceLength);
+
+                while (!Unsafe.AreSame(ref currentSearchSpace, ref searchSpaceEnd))
+                {
+                    char c = (char)currentSearchSpace;
+                    if (TNegator.NegateIfNeeded(state.Lookup.FastContains(c)))
+                    {
+                        return TResultMapper.ScalarResult(ref searchSpace, ref currentSearchSpace);
+                    }
+
+                    currentSearchSpace = ref Unsafe.Add(ref currentSearchSpace, 1);
+                }
+
+                return TResultMapper.NotFound;
+            }
+
+#pragma warning disable IntrinsicsInSystemPrivateCoreLibAttributeNotSpecificEnough // The behavior of the rest of the function remains the same if Avx2.IsSupported is false
+            if (Avx2.IsSupported && searchSpaceLength > 2 * Vector128<short>.Count)
+#pragma warning restore IntrinsicsInSystemPrivateCoreLibAttributeNotSpecificEnough
+            {
+#pragma warning disable IntrinsicsInSystemPrivateCoreLibAttributeNotSpecificEnough // The behavior of the rest of the function remains the same if Avx512BW.IsSupported is false
+                if (Vector512.IsHardwareAccelerated && Avx512BW.IsSupported && searchSpaceLength > 2 * Vector256<short>.Count)
+#pragma warning restore IntrinsicsInSystemPrivateCoreLibAttributeNotSpecificEnough
+                {
+                    Vector512<byte> asciiBitmap512 = state.AsciiBitmap512;
+                    Vector512<byte> secondBitmap512 = state.SecondBitmap512;
+                    Vector512<ushort> offset512 = Vector512.Create(state.Offset);
+
+                    if (searchSpaceLength > 2 * Vector512<short>.Count)
```

Comment:
@GrabYourPitchforks, in the UTF8 experiment you shared with me, you had code that, after having validated that Vector256 was hardware accelerated, aligned an address and then read a full vector, without concern for whether that vector read under or overread the target region. Is that safe to do on all platforms? It seems we could avoid some branching with similar techniques in many of our implementations that use vectorization.

Reply:
> Is that safe to do on all platforms?

As long as the data is pinned, otherwise GC can interrupt at any point and break the alignment assumption